packages/tx/src/baseTransaction.ts

```diff
@@ -298,6 +299,54 @@ export abstract class BaseTransaction<TransactionObject> {
     return tx
   }
 
+  /**
+   * Signs a transaction.
+   *
+   * Note that the signed tx is returned as a new object,
+   * use as follows:
+   * ```javascript
+   * const signer = (buffer: Buffer): Promise<{ signature, recid }>
+   * const signedTx = await tx.sign(signer)
+   * ```
+   */
+  signWithSigner(
+    signer: (msgHash: Buffer) => Promise<{ signature: Uint8Array; recid: number }>
+  ): Promise<TransactionObject> {
+    // Hack for the constellation that we have got a legacy tx after spuriousDragon with a non-EIP155 conforming signature
+    // and want to recreate a signature (where EIP155 should be applied)
+    // Leaving this hack lets the legacy.spec.ts -> sign(), verifySignature() test fail
+    // 2021-06-23
+    let hackApplied = false
+    if (
+      this.type === 0 &&
+      this.common.gteHardfork('spuriousDragon') &&
+      !this.supports(Capability.EIP155ReplayProtection)
+    ) {
+      this.activeCapabilities.push(Capability.EIP155ReplayProtection)
+      hackApplied = true
+    }
+
+    const msgHash = this.getMessageToSign(true)
+
+    const _processSignature = ({ signature, recid }: { signature: any; recid: number }) => {
+      const { r, s, v } = splitSignature(signature, recid)
+
+      const tx = this._processSignature(v, r, s)
+
+      // Hack part 2
+      if (hackApplied) {
+        const index = this.activeCapabilities.indexOf(Capability.EIP155ReplayProtection)
+        if (index > -1) {
+          this.activeCapabilities.splice(index, 1)
+        }
+      }
+
+      return tx
+    }
+
+    return signer(msgHash).then(({ signature, recid }) => _processSignature({ signature, recid }))
```

Comment:
The signer returns a recovery id of `0` or `1` but `_processSignature` requires it to be `26` or `27`.
https://github.com/ExodusMovement/assets/blob/main/ethereum/ethereum-lib/src/unsigned-tx/sign-hardware.js#L46-L50

Reply:
Oh, does `splitSignature` take care of that? 

https://github.com/ExodusMovement/ethereumjs-monorepo/blob/b6474868069c33ebf42a717601af186c7cb4f1a9/packages/util/src/signature.ts#L54

Reply:
Yup.